### PR TITLE
Join table for visits

### PR DESCRIPTION
**Pull request by zamcham**
Added some many-to-many relationships and write more complex queries to use them to answer questions. Results:

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -27,3 +27,16 @@ id SERIAL PRIMARY KEY,
...
+  name VARCHAR(255),
+  age INTEGER,
+  date_of_graduation DATE
+);
+
+CREATE TABLE specializations (
+  vet_id INTEGER REFERENCES vets(id),
+  species_id INTEGER REFERENCES species(id),
+  PRIMARY KEY (vet_id, species_id)
+);
```
Comment (afanimartin) [inline schema.sql:43]: - Please add a new table `Visits`. Should have a schema like below
```
   CREATE TABLE visits (
   animal_id int,
   vet_id int,
   visit_date date -- another name is acceptable
   );
```

File: data.sql [hunk 2]
```diff
@@ -63,3 +63,95 @@ WHERE name IN ('Charmander', 'Squirtle', 'Blossom');
...
+INSERT INTO visits (animal_id, vet_id, visit_date)
+VALUES ((SELECT id FROM animals WHERE name = 'Boarmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2020-08-03');
+
+-- Blossom visited Stephanie Mendez on May 24th, 2020
+INSERT INTO visits (animal_id, vet_id, visit_date)
+VALUES ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2020-05-24');
+
+-- Blossom visited William Tatcher on Jan 11th, 2021
+INSERT INTO visits (animal_id, vet_id, visit_date)
+VALUES ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2021-01-11');
```
Comment (afanimartin) [inline data.sql:169]: - Please insert the following data into `Vets` table
```
   INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('William Tatcher', 45, '2000-04-23');

   INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Maisy Smith', 26, '2019-01-17');

   INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Stephanie Mendez', 64, '1981-05-04');

   INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Jack Harkness', 38, '2008-06-08');
`

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @zamcham, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Everything that creates or change the tables structure is saved in the schema.sql file.
- Everything that populates tables with data is saved in the data.sql file.
- Everything that is a query that helps to analyze data is saved in the queries.sql file.
- Professional readme template used

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by christianbiring1 — APPROVED** [review]
Hi @zamcham , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 

## Highlights
- [ ] Great work on implementing the changes as required by the previous reviewer :+1: 
- [ ] Great job on providing the correct queries to answer all the asked questions :1st_place_medal: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**